vllm/model_executor/layers/rejection_sampler.py

```diff
@@ -77,24 +82,33 @@ def forward(
         """
         # Only perform shape/dtype/device checking in strict mode, as it adds
         # overhead.
+        target_probs = target_with_bonus_probs[:, :-1]
```

Comment:
nit: the `raise_if_incorrect_input` function should check that the input is correct, before any modification. otherwise it is less helpful as it becomes more like an assertion on internal state, not on correct input